tests/test_schema_validation.py

```diff
@@ -0,0 +1,24 @@
+import unittest
+
+from openapi.helpers import validate_schema
+from openapi.exceptions import ValidationError
+
+
+class SchemaValidationTestCase(unittest.TestCase):
+
+    def setUp(self) -> None:
+        self.schema = {
+            'type': 'string',
+            'pattern': '^my string$',
+        }
+
+    def test_valid_schema_validation(self):
+        self.assertIsNone(validate_schema(self.schema, 'my string'))
+
+    def test_invalid_schema_validation(self):
+        with self.assertRaises(ValidationError):
```

Comment:
Maybe it is worth testing for `ValidationError.errors` since there is some code to add them.

Reply:
Great! I was able to detect an issue with the validation function.